### ISSUE TITLE
[next] Both `src` and `worker` can be used and no error is shown

### ISSUE THREAD

**Issue by WebReflection** [pyscript/pyscript#1730]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

We recently landed this MR https://github.com/pyscript/pyscript/pull/1724 but there's one use case that's not tackled: a script has both `src` and `worker` attributes.

I think we should follow that very same PyScript error code and throw when we have both `src` and `worker`, because currently `worker` has priority (it doesn't bootstrap the engine on the main thread so it can't be de-prioritized) and it's not obvious that `src` is ignored.

**Comment by ntoll** [comment]
This arose because I'm writing up and checking the behaviour of PyScript. Here's what I wrote, only to realise it's wrong.

> Danger
>
> Do not use both the `src` and `worker` attributes at the same time.
>
> If you want to run code on both the main thread and in a worker, be explicit and use separate tags: one with a `src` attribute referencing code to be run on the main thread, the other with a `worker` attribute referencing code to be run in a worker.

**Comment by WebReflection** [comment]
> PyScript will raise an exception in the browser console.

I would rephrase it as

> PyScript will raise an exception

let's not specify where exceptions are raised as the error plugin, enabled by default, shows error on the page ... true that errors might be logged in console too, but we don't need to specify the where, imho.

**Comment by antocuni** [comment]
> I think we should follow that very same PyScript error code and throw when we have both src and worker, because currently worker has priority (it doesn't bootstrap the engine on the main thread so it can't be de-prioritized) and it's not obvious that src is ignored.

+1 for throwing an error in this case (and, as @WebReflection says, *where* this error is displayed will depends on the error plugin).

However, let me propose a slightly different syntax:
```html
<script type="py">                      # main thread, inline code
<script type="py" src="foo.py">         # main thread, external code
<script type="py" worker>               # worker, inline code
<script type="py" src="foo.py" worker>  # worker, external code
```

i.e., `worker="foo.py"` is no longer supported and it simply become a boolean flag to indicate in which thread the code is supposed to run.

**NOTE**: I don't know:
- whether it's a good idea
- how hard it is to implement this in polyscript/pyscript
- whether we should consider it for the upcoming release or it's too late and in that case better to wait for the next

But I thought it's an idea worth exploring

**Comment by ntoll** [comment]
@antocuni hmm... I like your proposal of `worker` as a flag. It's just a flag, simple to understand and no need for throwing exceptions. @WebReflection any problems with this approach?

**Comment by WebReflection** [comment]
isn't this **exactly** what I wrote in our internal discussion? :sweat_smile: 

as I need to change *polyscript* anyway because it bails out prematurely to avoid bootstrapping *pyodide* on main when a `worker` has a value, I am OK with that resolution.

This requires more work but if I can just go ahead it should be done by the end of today (I'll finish at 6pm CET).

**Comment by ntoll** [comment]
I am an idiot. 🤦 🤣

**Comment by ntoll** [comment]
Let's go with the @WebReflection / @antocuni suggestion! Apologies for misreading.

**Comment by WebReflection** [comment]
@antocuni @ntoll bear in mind this requires fundamental changes in *polyscript* as breaking release unless I offer this approach only to custom types but that feels *yac* to me ... also worth saying that the ambiguous code case would be simplified by just checking `src` attribute so ... it should be a win-win at the end but I don't know if we have already tests to change for this too.

**Comment by WebReflection** [comment]
related https://github.com/pyscript/polyscript/issues/43

**Comment by ntoll** [comment]
I think this is definitely the right thing to do. So much less confusing. Better to update this NOW before we release to the world. :+1:

**Comment by antocuni** [comment]
+1 for updating it now 👍

**Comment by WebReflection** [comment]
@antocuni ... see? [this one](https://github.com/pyscript/pyscript/pull/1732) didn't close a thing :shrug:
